tests/test_utils.py

```diff
@@ -31,26 +31,44 @@ def test_eqpb_2d(self):
         number of bin edges. Test this for both orderings.'''
         n_data = 12
         n_partitions = [2, 3]
-        data_sample = np.linspace(0, 1, n_data)
-        data_sample = np.vstack([data_sample for i in range(2)]).T
-        reference_sample = np.linspace(0, 1, int(10 * n_data))
-        reference_sample = np.vstack([reference_sample for i in range(2)]).T
+        for binning in ['log', 'lin']:
+            if binning == 'lin':
+                data_sample = np.linspace(0, 1, n_data)
+                reference_sample = np.linspace(0, 1, int(10 * n_data))
+            elif binning == 'log':
+                data_sample = np.logspace(0, 1, n_data)
+                reference_sample = np.linspace(1, 10, int(10 * n_data))
+            if binning == 'lin':
+                reference_sample_weights_l = [None, np.ones(10 * n_data)]
+            elif binning == 'log':
+                reference_sample_weights = np.linspace(1, 10, int(10 * n_data))
+                reference_sample_weights = np.log(reference_sample_weights)
+                reference_sample_weights = np.hstack([0, reference_sample_weights])
+                reference_sample_weights = np.diff(reference_sample_weights)
+                reference_sample_weights_l = [reference_sample_weights]
```

Comment:
Not sure if I understand correctly what you're aiming to do here. It looks like you want to adjust the weights such that you get the same binning as in the lin case but I get different values for both.. Also, you should then also check this somewhere, i.e. check that in the lin case weights=None and weights=np.ones gives the same bin_edges and maybe adding the weights for log you can also obtain the same but not sure if this is actually what you wanted to do as I said..😄 If this becomes too long you could also just write a second test, no need to fit everything into one test ;)

Reply:
I think I was absurd when I  wrote those code. What I was going to do here is just make the weights represent some histograms. In this case, it is a linear and log-like histogram.